integrationTests/vm/arwen/arwenvm/deployment/deploy_test.go

```diff
@@ -0,0 +1,111 @@
+package deployment
+
+import (
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm/arwen"
+	"github.com/ElrondNetwork/elrond-go/state"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestScDeployShouldManageCorrectlyTheCodeMetadata(t *testing.T) {
+	senderAddressBytes := []byte("12345678901234567890123456789012")
+	senderNonce := uint64(0)
+	senderBalance := big.NewInt(100000000)
+
+	testContext, err := vm.CreatePreparedTxProcessorAndAccountsWithVMs(
+		senderNonce,
+		senderAddressBytes,
+		senderBalance,
+		config.EnableEpochs{
+			IsPayableBySCEnableEpoch: 1,
+		},
+	)
+	require.Nil(t, err)
+	defer testContext.Close()
+
+	t.Run("payable by SC is not active", func(t *testing.T) {
+		testContext.EpochNotifier.CheckEpoch(&block.Header{Epoch: 0})
+
+		contractAddress := deployDummySCReturningContractAddress(t, testContext, senderAddressBytes, "FFFF")
+
+		expectedCodeMetadata := vmcommon.CodeMetadata{
+			Payable:     true,
+			PayableBySC: false,
+			Upgradeable: true,
+			Readable:    true,
+		}
+
+		assert.Equal(t, expectedCodeMetadata.ToBytes(), getCodeMetadata(t, testContext.Accounts, contractAddress))
+	})
+	t.Run("payable by SC is active", func(t *testing.T) {
+		testContext.EpochNotifier.CheckEpoch(&block.Header{Epoch: 1})
+
+		contractAddress := deployDummySCReturningContractAddress(t, testContext, senderAddressBytes, "FFFF")
+
+		expectedCodeMetadata := vmcommon.CodeMetadata{
+			Payable:     true,
+			PayableBySC: true,
+			Upgradeable: true,
+			Readable:    true,
+		}
+
+		assert.Equal(t, expectedCodeMetadata.ToBytes(), getCodeMetadata(t, testContext.Accounts, contractAddress))
+	})
+	t.Run("payable by SC is active, wrong number of bytes in code metadata argument", func(t *testing.T) {
```

Comment:
can you add some comment why all is false ?

Reply:
done

Reply:
Should the deployContract work? It just deployed a contract that is not upgradeable, not readable, not payable and not payableBySC allthough the contractMetaData is invalid?

Reply:
in this case we can leave it as it is. If someone misses a deploy operation, she can always redeploy a contract. Of course, we can add a check under the same activation flag flag returning error on the sc processor. What do you think, @sasurobert ?

Reply:
done, for indirect depoyment